patches/tModLoader/Terraria.ModLoader/ModWorld.cs

```diff
@@ -108,6 +108,88 @@ public virtual void ResetNearbyTileEffects()
 		{
 		}
 
+		/// <summary>
+		/// Use this if you want to do something before anything in the World gets updated.
+		/// Gets called after UI updates, but before anything in the World (Players, NPCs, Projectiles, Tiles) gets updated.
+		/// <para />
+		/// When <see cref="Main.autoPause" /> is true or <see cref="Main.FrameSkipMode" /> is 0 or 2, the game may do a partial update. This means that it only updates menus and some animations, but not the World or Entities. This hook - and every hook after it - only gets called on frames with a full update.
+		/// <para />
+		/// If you are looking too hook a later part of the update process, see <see cref="MidUpdatePlayerNPC" />.
+		/// </summary>
+		public virtual void PreUpdateAnything()
+		{
+		}
+
+		/// <summary>
+		/// Gets called immedeately after all Players are updated, but before any NPCs get updated.
+		/// <para />
```

Comment:
*immediately

Reply:
Sometimes I just can't even, the **i** is literally pronounced the same as the **e** right before it. Why do you have to do this, English?